### PR TITLE
[Relay][Pass][Docs] Update the doc for adding a Relay pass to mention the pass infra

### PR DESCRIPTION
**Pull request by slyubomirsky**
Follow-up to issue #3480 and PR #3515: I've updated the existing documentation on writing a Relay pass to reflect the use of the pass infrastructure and to reference the pass infrastructure docs.

Please, if you can, review for accuracy, style, and correctness @zhiics @weberlo @joshpoll and any others who may be knowledgeable

### CODE REVIEW

File: docs/dev/relay_add_pass.rst [hunk 1]
```diff
@@ -327,32 +325,82 @@ all of the arguments are constant (using ``ConstantChecker``). Evaluating the
...
+allows for registering the optimization level of the pass (in this case, 2), which can
+be used to group together passes based on their general utility, a name for the pass,
+and any dependencies for the pass. A pass's dependencies are given as a list of any passes
+whose results are necessary to be able to run the current pass. ``FoldConstant`` does not
+have any dependencies, but many Relay passes do depend on having type information,
+so ``InferType`` is a common dependency; others may depend on the program's being in
+A-normal form, via the ``ToANormalForm`` pass.
+
+Note that the ``PassContext`` object contains information a pass uses for
+error reporting and configuration options; ``FoldContext`` does not need
```
Comment (zhiics) [inline docs/dev/relay_add_pass.rst]:
```suggestion
error reporting and configuration options; ``FoldConstant`` does not need
```

File: docs/dev/relay_pass_infra.rst [hunk 2]
```diff
@@ -15,7 +15,9 @@
     specific language governing permissions and limitations
     under the License.
 
-Relay Pass Infra
+.. _relay-pass-infra:
+
+Relay Pass Infrastructure
 ==================================
```
Comment (zhiics) [inline docs/dev/relay_pass_infra.rst]: Align `=` with the header?

File: docs/dev/relay_add_pass.rst (hunk 1 above)
Comment (slyubomirsky) [reply docs/dev/relay_add_pass.rst]: Good catch

File: docs/dev/relay_pass_infra.rst (hunk 2 above)
Comment (slyubomirsky) [reply docs/dev/relay_pass_infra.rst]: Will do

**Pull request merged**